tests/chart_tests/test_byo_sa.py

```diff
@@ -250,6 +252,8 @@ def test_default_serviceaccount_names(template_name):
     default_serviceaccount_names_overrides = {"global": {"rbacEnabled": False}, "postgresql": {"serviceAccount": {"enabled": True}}}
     if "nginx-dp-deployment" in template_name:
         default_serviceaccount_names_overrides["global"]["plane"] = {"mode": "data"}
+    if "fluentd-daemonset" in template_name:
+        default_serviceaccount_names_overrides["global"]["logging"] = {"collector": "fluentd"}
```

Comment:
Because these are exclusive `if`s we should turn this into a match ... case block
```suggestion
match template_name:
    case _ if "nginx-dp-deployment" in template_name:
        default_serviceaccount_names_overrides["global"]["plane"] = {"mode": "data"}
    case _ if "fluentd-daemonset" in template_name:
        default_serviceaccount_names_overrides["global"]["logging"] = {"collector": "fluentd"}
```

Reply:
Also maybe this should be `vector-daemonset` ?